### ISSUE TITLE
"New cards ignore review limit" doesn't apply to simulator

### ISSUE THREAD

**Issue by brishtibheja** [ankitects/anki#3654]
The setting "New cards ignore review limit" doesn't apply to the simulator. 

I noticed when "Review limit" is 0 but "New card limit" is a non-zero number, the simulator shows 0 reviews for everyday, but in my case I have the said option turned on.

**Comment by L-M-Sherlock** [comment]
Oops. I guess it's caused by 

https://github.com/open-spaced-repetition/fsrs-rs/blob/8cf1de85a54bea3280366b795a3f7ebd4ee3cd9c/src/optimal_retention.rs#L241

I will fix it soon.

**Comment by brishtibheja** [comment]
@L-M-Sherlock Have you fixed this? The `new cards ignore review limit` toggle should work for sim.

**Comment by L-M-Sherlock** [comment]
Nope. If the toggle should work for sim, I have to introduce a breaking change. Instead, I make the sim always behave as `new cards ignore review limit` is enabled.

**Comment by brishtibheja** [comment]
This is resolved after https://github.com/ankitects/anki/pull/3707.
